src/modules/checkboxes.js

```diff
@@ -0,0 +1,45 @@
+// checkboxes.js
+import { saveTasksToLocalStorage } from './taskFunctions.js';
+
+// Add checkbox functionality to tasks
+export function addCheckboxFunctionality(taskList) {
+  taskList.addEventListener('change', (event) => {
+    if (event.target.classList.contains('task-checkbox')) {
+      const listItem = findParentListItem(event.target);
+      if (listItem) {
+        const taskText = listItem.querySelector('.task-item span'); // Updated selector
+        taskText.classList.toggle('completed', event.target.checked);
+
+        saveTasksToLocalStorage(taskList);
+      }
+    }
+  });
+}
```

Comment:
- [ ] The status update functionality should toggle the `completed` key of every task object between `true` and `false`. Currently, each of your task's objects doesn't have a `completed` key. I would advise you to include it when creating a task. Then when a user clicks on the checkbox for each task, it should update the `completed` key to `true` or `false`. Kindly implement this change. 👍 😎 

![image](https://github.com/DevKaranJ/ToDo-list/assets/77106962/7cac0f61-71a9-4c53-8369-80361d826542)